### PR TITLE
[CL: Positions] Add GetPositionsByUserAddress for CL user positions

### PR DESCRIPTION
**Pull request by stackman27**
Closes: [#4115](https://github.com/osmosis-labs/osmosis/issues/4115)

## What is the purpose of the change

Currently in CL pools we Getpositions by poolid, userid, lowerTick, upperTick. We also would like to retrieve all current positions across many pools.

## Brief Changelog

- Added new keeper method that retrieves users position

## Testing and Verifying
added tests 

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: x/concentrated-liquidity/position.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/position.go:106]: Isn't this the same function from the file I sent you? If so, we should just point to that instead of code duplication.

File: x/concentrated-liquidity/position_test.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/position_test.go:187]:
```suggestion
func (s *KeeperTestSuite) TestGetAllUserPositions() {
```

File: x/concentrated-liquidity/position_test.go
Comment (czarcas7ic) [inline x/concentrated-liquidity/position_test.go:243]:
```suggestion
				// if position does not exist this errors
```

**Review by czarcas7ic — APPROVED** [review]

File: x/concentrated-liquidity/position.go
Comment (stackman27) [reply x/concentrated-liquidity/position.go:106]: yup it is, i didn't branch off your PR, so this function does exist right now. Still wondering if we should wait for your PR to get merged before merging this one

Comment (p0mvn) [reply x/concentrated-liquidity/position.go:106]: Let's try merging this sooner rather than later please because this is blocking e2e work

File: x/concentrated-liquidity/position.go
Comment (p0mvn) [inline x/concentrated-liquidity/position.go:100]:
```suggestion
// ParsePositionFromBz parses bytes into a position struct. Returns a parsed position and nil on success.
// Returns error if bytes length is zero or if fails to parse the given bytes into the position struct.
func ParsePositionFromBz(bz []byte) (position model.Position, err error) {
```

**Review by p0mvn — APPROVED** [review]

**Review by RusAkh — APPROVED** [review]

**Comment by stackman27** [comment]
Rebased and ready for merge now!

**Pull request merged**
